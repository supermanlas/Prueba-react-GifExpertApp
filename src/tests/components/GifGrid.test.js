import React from 'react';
import { shallow } from "enzyme"
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs'); // Fingir llamada al archivo y controlar la información que este responda

describe('Prueba componente <GifGrid />', () => {
    
    const category = 'Detective Conan'

    test('Debe de mostrar componente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect( wrapper ).toMatchSnapshot();
        
    })
    
    test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {

        const gifs = [{
            id:'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'cualquier cosa'
        },{
            id:'123',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'cualquier cosa'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow(<GifGrid category={category} />);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false); //Evalúa si existe algún 'p'
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length ); // Revisa cuantos elementos GifGridItem existen

        
    })
    
    
})
