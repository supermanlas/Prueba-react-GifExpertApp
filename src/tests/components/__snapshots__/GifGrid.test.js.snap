// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba componente <GifGrid /> Debe de mostrar componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn "
  >
    Detective Conan
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Prueba componente <GifGrid /> Debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn "
  >
    Detective Conan
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;
